lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,427 @@
+
+    expect(onSearch).not.toHaveBeenCalled();
+    await userEvent.type(inputElement, ' ');
+    expect(onSearch).not.toHaveBeenCalled();
+    await userEvent.type(inputElement, 'f');
+    expect(onSearch).toHaveBeenCalledWith('foof');
+  });
+
+  it('adds the nx-text-input--long class to the div if set', function() {
+    expect(rtlRender().container.querySelector('.nx-combobox')).not.toHaveClass('nx-text-input--long');
+    expect(rtlRender({ className: 'nx-text-input--long' }).container.querySelector('.nx-combobox'))
+        .toHaveClass('nx-text-input--long');
+  });
+
+  it('adds the nx-text-input--short class to the div if set', function() {
+    expect(rtlRender().container.querySelector('.nx-combobox')).not.toHaveClass('nx-text-input--short');
+    expect(rtlRender({ className: 'nx-text-input--short' }).container.querySelector('.nx-combobox'))
+        .toHaveClass('nx-text-input--short');
+  });
+
+  it('passes the disabled prop to the input and buttons', function() {
+    const { getByRole, rerender } = rtlRender(),
+        inputElement = getByRole('combobox');
+
+    expect(inputElement).toHaveProperty('disabled', false);
+
+    rerender(<NxCombobox {...minimalProps} disabled={undefined} />);
+    expect(inputElement).toHaveProperty('disabled', false);
+
+    rerender(<NxCombobox {...minimalProps} disabled={false} />);
+    expect(inputElement).toHaveProperty('disabled', false);
+
+    rerender(<NxCombobox {...minimalProps} matches={[{ id: '1', displayName: 'Foo' }]} disabled={false} />);
+    expect(getByRole('option', { hidden: true })).toHaveProperty('disabled', false);
+
+    rerender(<NxCombobox {...minimalProps} matches={[{ id: '1', displayName: 'Foo' }]} disabled={true} />);
+    expect(getByRole('option', { hidden: true })).toHaveProperty('disabled', true);
+  });
+
+  it('renders a dropdown with the nx-combobox__menu class', async function() {
+    const { getByRole } = rtlRender({ matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }] }),
+        dropdownElement = getByRole('listbox', { hidden: true });
+
+    expect(dropdownElement).toHaveClass('nx-combobox__menu');
+  });
+
+  it('sets the listbox role on the dropdown', function() {
+    const { getByRole } = rtlRender({ matches: [{ id: '1', displayName: 'Foo' }] }),
+        dropdownElement = getByRole('listbox', { hidden: true });
+
+    expect(dropdownElement).toHaveAttribute('role', 'listbox');
+  });
+
+  it('sets an id on the dropdown and references it in the combobox input aria-controls', function() {
+    const { getByRole } = rtlRender({ matches: [{ id: '1', displayName: 'Foo' }] }),
+        inputElement = getByRole('combobox'),
+        dropdownElement = getByRole('listbox', { hidden: true });
+
+    expect(dropdownElement).toHaveAttribute('id');
+    expect(inputElement).toHaveAttribute('aria-controls', dropdownElement.id);
+  });
+
+  it('renders an alert dropdown with nx_combobox__alert class', function() {
```

Comment:
Isn't the alert only present in some cases?

Reply:
Yes, it is rendered when there is loading, loadError or empty(No matches).